aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -82,14 +70,19 @@ void binary_cross_entropy_backward_out_kernel(Tensor& grad_input, const Tensor&
 namespace at { namespace native {
 
 Tensor kl_div_backward_cuda(const Tensor& grad, const Tensor& input, const Tensor& target, int64_t reduction) {
-  auto grad_input = at::zeros_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  Tensor grad_expand = grad.expand_as(input);
+  auto grad_input = at::empty_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+  TensorIterator iter;
+  iter.add_output(grad_input);
+  iter.add_input(target);
+  iter.add_input(grad);
+  iter.build();
   AT_DISPATCH_FLOATING_TYPES_AND_HALF(input.scalar_type(), "kl_div_backward_cuda", [&]() {
-    kl_div_backward_kernel<scalar_t>(grad_input, target, grad_expand);
+    scalar_t inv = (reduction == at::Reduction::Mean) ? scalar_t(1.0 / input.numel()) : scalar_t(1.0);
```

Comment:
This might introduce slowdown due to additional operation. Can you please add benchmarks to the PR.

Reply:
It is actually not a slowdown, but I fused the
```C++
  if (reduction == at::Reduction::Mean) {	
    return grad_input / input.numel();	
  }
```
into the iterator, so it should be faster. But I will do a benchmark.

Reply:
I recognize the intend, but fusion requires two kernels for  `reduction == at::Reduction::Mean` and without it. Your version _might_ force additional multiplication call (depends from compiler optimization).